man/man4/zfs.4

```diff
@@ -680,6 +680,22 @@ Minimum time "prescient prefetched" blocks are locked in the ARC.
 These blocks are meant to be prefetched fairly aggressively ahead of
 the code that may use them.
 .
+.It Sy zfs_arc_prune_delay_ms Ns = Ns Sy 2000 Ns ms Po 2s Pc Pq ulong
+If nothing is able to be pruned from a superblock, delay our next attempt at
+pruning this superblock for some milliseconds to avoid continually trying to
+do something that can't be done.
+Under pressure, this can save thousands of scans a second per superblock.
+The delay time isn't critical, it needs to be long enough to avoid
+excessive useless work, and short enough that skipping the prune on this
+particular superblock isn't going to cause problems.
+.Pp
+The parameter can be set to 0 (zero) to disable the delay,
+and only applies on Linux.
+.Pp
+The number of prunes and delayed prune attempts will be available in
+.Pa /proc/spl/kstat/zfs/ Ns Ao Ar pool Ac Ns Pa / Ns Ao Ar objset Ac Ns Pa :prunes and
```

Comment:
break before and

Reply:
Will fix in next push

Reply:
By the way, is this a reasonable way of referring to specific fields within a kstat file? I couldn't find any other obvious example.

Reply:
This decodes to, what, `/proc/spl/kstat/zfs/<pool>/<objset>:prunes`, so `cat /proc/spl/kstat/zfs/filling/133182:prunes` will work as I'd expect it to? I don't remember there being any kstat paths in the manuals but this is good syntax, yeah.

Reply:
No, can't `cat` that whole path. It looks like:
```
$ cat /proc/spl/kstat/zfs/pool/objset-0x1d6f
1037 1 0x01 9 2704 7659484348610 84937346817053
name                            type data
dataset_name                    7    pool/fubar
writes                          4    239129
nwritten                        4    80673747
reads                           4    738074
nread                           4    26010672
nunlinks                        4    135
nunlinked                       4    135
prunes                          4    616
prunes_delayed                  4    173
```

Reply:
Right; in that case I'd go for something like 
```
The number of prunes and delayed prune attempts will be available in the
.Va prunes
and
.Va prunes_delayed
fields of
.Pa /proc/spl/kstat/zfs/ Ns Ao Ar pool Ac Ns Pa /objset-0x Ns Aq Ar objset .
```

Reply:
Good stuff - it'll be in the next push